components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/drop-elements.js

```diff
@@ -1404,37 +1404,18 @@ define(['require', 'log', 'lodash', 'jquery', 'partition', 'stream', 'query', 'f
             var self = this;
             var oldFaultStreamName = Constants.FAULT_STREAM_PREFIX + previouslySavedStreamName;
             var oldFaultStreamElementId = "";
-            self.configurationData.getSiddhiAppConfig().getStreamList().forEach(function(s) {
+            self.configurationData.getSiddhiAppConfig().getStreamList().forEach(function (s) {
                 if (oldFaultStreamName === s.getName()) {
                     oldFaultStreamElementId = s.getId();
                 }
             });
 
             if (oldFaultStreamElementId != "") {
                 self.configurationData.getSiddhiAppConfig().removeStream(oldFaultStreamElementId);
-                $( "#" + oldFaultStreamElementId ).remove();
+                $("#" + oldFaultStreamElementId).remove();
             }
             if (stream.hasFaultStream()) {
-                //remove existing connectionDots
-                $('#' + stream.getId() + '-in').remove();
-                $('#' + stream.getId() + '-out').remove();
-
-                var connection1 = $('<div class="connectorInStreamForErrorStream">').attr('id', stream.getId() + "-in").
-                    addClass('connection');
-                var connection2 = $('<div class="connectorOutForErrorStream">').attr('id', stream.getId() + "-out").
-                    addClass('connection');
-                $('#' + stream.getId()).append(connection1);
-                $('#' + stream.getId()).append(connection2);
-
-                jsPlumbInstance.makeTarget(connection1, {
-                    deleteEndpointsOnDetach: true,
-                    anchor: 'Left'
-                });
-
-                jsPlumbInstance.makeSource(connection2, {
-                    deleteEndpointsOnDetach: true,
-                    anchor: 'Right'
-                });
+                self.deleteAndConnectConnection(stream.getId());
```

Comment:
shall we change the method name to removeAndRecreateConnection

Reply:
fixed in b89e75c6555fa5ba9f04dfc5e092dd39e98edcd1